app/controllers/pdl.py

```diff
@@ -803,6 +753,245 @@ def get_program_id(program_name):
     return program.id
 
 
+def generate_allprograms_response(start_year, end_year):
```

Comment:
One thing that I would like suggest as an improvement is to use SQLAlchemy (without direct SQL queries) and do join on the `titles` table. That way we will have less number of lines of code and we can avoid hardcoding the query and the title names. 

Since this change needs to be released this week, maybe it can be a follow up task as part of code clean up.